tests/test_grids.py

```diff
@@ -645,24 +646,53 @@ def test_cgrid_indexing(pset_mode, mode, gridindexingtype):
+                           [-np.sin(alpha), np.cos(alpha)]])
+        lons, lats = np.meshgrid(lon, lat)
+        rotated = np.einsum('ji, mni -> jmn', rotmat, np.dstack([lons, lats]))
+        return rotated[0], rotated[1]
+
+    if coordtype == 'rectilinear':
+        alpha = 0
+    elif coordtype == 'curvilinear':
+        alpha = 15*np.pi/180
+        lon, lat = rotate_coords(lon, lat, alpha)
+
     def calc_r_phi(ln, lt):
         return np.sqrt(ln ** 2 + lt ** 2), np.arctan2(ln, lt)
 
-    def calculate_UVR(lat, lon, dx, dy, omega):
-        U = np.zeros((lat.size, lon.size), dtype=np.float32)
-        V = np.zeros((lat.size, lon.size), dtype=np.float32)
-        R = np.zeros((lat.size, lon.size), dtype=np.float32)
-        for i in range(lon.size):
-            for j in range(lat.size):
-                r, phi = calc_r_phi(lon[i], lat[j])
-                R[j, i] = r
-                r, phi = calc_r_phi(lon[i] + isign * dx / 2, lat[j])
-                V[j, i] = -omega * r * np.sin(phi)
-                r, phi = calc_r_phi(lon[i], lat[j] + isign * dy / 2)
-                U[j, i] = omega * r * np.cos(phi)
-        return U, V, R
-
-    U, V, R = calculate_UVR(lat, lon, dx, dy, omega)
+    if coordtype == 'rectilinear':
+        def calculate_UVR(lat, lon, dx, dy, omega, alpha):
+            U = np.zeros((lat.size, lon.size), dtype=np.float32)
+            V = np.zeros((lat.size, lon.size), dtype=np.float32)
+            R = np.zeros((lat.size, lon.size), dtype=np.float32)
+            for i in range(lon.size):
+                for j in range(lat.size):
+                    r, phi = calc_r_phi(lon[i], lat[j])
+                    R[j, i] = r
+                    r, phi = calc_r_phi(lon[i] + isign * dx / 2, lat[j])
+                    V[j, i] = -omega * r * np.sin(phi)
+                    r, phi = calc_r_phi(lon[i], lat[j] + isign * dy / 2)
+                    U[j, i] = omega * r * np.cos(phi)
+            return U, V, R
+    elif coordtype == 'curvilinear':
+        def calculate_UVR(lat, lon, dx, dy, omega, alpha):
+            U = np.zeros(lat.shape, dtype=np.float32)
+            V = np.zeros(lat.shape, dtype=np.float32)
+            R = np.zeros(lat.shape, dtype=np.float32)
+            for i in range(lat.shape[1]):
+                for j in range(lat.shape[0]):
+                    r, phi = calc_r_phi(lon[j, i], lat[j, i])
+                    R[j, i] = r
+                    r, phi = calc_r_phi(lon[j, i] + isign * (dx / 2) * np.cos(alpha), lat[j, i] - isign * (dx / 2) * np.sin(alpha))
+                    V[j, i] = np.sin(alpha) * (omega * r * np.cos(phi)) + np.cos(alpha) * (-omega * r * np.sin(phi))
+                    r, phi = calc_r_phi(lon[j, i] + isign * (dy / 2) * np.sin(alpha), lat[j, i] + isign * (dy / 2) * np.cos(alpha))
+                    U[j, i] = np.cos(alpha) * (omega * r * np.cos(phi)) - np.sin(alpha) * (-omega * r * np.sin(phi))
+            return U, V, R
```

Comment:
I've now merged the two tests. I didn't opt for simply having an `alpha` argument, as I think it's also good to differentiate the the case when `lon` and `lat` are 1D or 2D arrays, as I guess this is the biggest way in which rectilinear versus curvilinear grid handling differ.